janus_core/calculations/geom_opt.py

```diff
@@ -172,90 +62,221 @@ def optimize(  # pylint: disable=too-many-arguments,too-many-locals,too-many-bra
-
-    # Calculate current maximum force
-    if filter_func is not None:
-        max_force = linalg.norm(filtered_struct.get_forces(), axis=1).max()
-    else:
-        max_force = linalg.norm(struct.get_forces(), axis=1).max()
-
-    if logger:
-        logger.info(message)
-        logger.info("Max force: %.6f", max_force)
-
-    if not converged:
-        warnings.warn(
-            f"Optimization has not converged after {steps} steps. "
-            f"Current max force {max_force} > target force {fmax}"
+        self.write_kwargs.setdefault(
+            "filename",
+            self._build_filename("opt.extxyz").absolute(),
         )
 
-    # Write out optimized structure
-    output_structs(
-        struct,
-        write_results=write_results,
-        write_kwargs=write_kwargs,
-    )
-
-    # Reformat trajectory file from binary
-    if traj_kwargs:
-        traj = read(opt_kwargs["trajectory"], index=":")
+        if traj_kwargs and "filename" not in traj_kwargs:
+            raise ValueError("'filename' must be included in `traj_kwargs`")
+
+        if traj_kwargs and "trajectory" not in opt_kwargs:
+            raise ValueError(
+                "'trajectory' must be a key in `opt_kwargs` to save the trajectory."
+            )
+
+        if log_kwargs and "filename" not in log_kwargs:
+            raise ValueError("'filename' must be included in `log_kwargs`")
+
+        log_kwargs.setdefault("name", __name__)
+        self.logger = config_logger(**log_kwargs)
+        self.tracker = config_tracker(self.logger, **tracker_kwargs)
+
+        self.set_optimizer()
+
+    def set_optimizer(self) -> None:
+        """Set optimizer for geometry optimization."""
+
+        self._set_functions()
+        if self.logger:
+            self.logger.info("Using optimizer: %s", self.optimizer.__name__)
+
+        if self.filter_func is not None:
+            if "scalar_pressure" in self.filter_kwargs:
+                self.filter_kwargs["scalar_pressure"] *= units.GPa
+            self.filtered_struct = self.filter_func(self.struct, **self.filter_kwargs)
+            self.dyn = self.optimizer(self.filtered_struct, **self.opt_kwargs)
+            if self.logger:
+                self.logger.info("Using filter: %s", self.filter_func.__name__)
+                if "hydrostatic_strain" in self.filter_kwargs:
+                    self.logger.info(
+                        "hydrostatic_strain: %s",
+                        self.filter_kwargs["hydrostatic_strain"],
+                    )
+                if "constant_volume" in self.filter_kwargs:
+                    self.logger.info(
+                        "constant_volume: %s", self.filter_kwargs["constant_volume"]
+                    )
+                if "scalar_pressure" in self.filter_kwargs:
+                    self.logger.info(
+                        "scalar_pressure: %s GPa",
+                        self.filter_kwargs["scalar_pressure"] / units.GPa,
+                    )
```

Comment:
Potentially, but I think I'd probably rather avoid unnecessary log statements, and it can also be useful to use the existence of the log statement as an indication that it has been set (either explicitly, or somewhere within the code).